packages/base/src/blockchainSchema.ts

```diff
@@ -0,0 +1,17 @@
+import { JsonObject, JsonValue } from "./jsonSchema";
+
+export type Hash = string;
+export type Amount = string;
+export type Address = Hash;
+export type UserId = Hash;
+
+
+interface TxRefBase<TBlockNumber extends JsonValue> extends JsonObject {
+  blockNumber?: TBlockNumber;
+  blockHash?: Hash;
+  txHash?: Hash;
+  sender?: UserId;
+}
```

Comment:
Having the same structure for `TxRef` is fine, but I don't like that all properties are optional what forces consumer to check whether they all set. I suggest rather:
```typescript

interface TxRefBase<TBlockNumber extends JsonValue> extends JsonObject {
  chain: string;
  blockNumber: TBlockNumber;
  blockHash: Hash;
  txHash: Hash;
  sender: UserId;
}

export type TxRef = TxRefBase<number>;
export type TxRefStr = TxRefBase<string>;
```

And every stream/protocol can decide how it's `TxRef` should look like, so every stream has it's strongly typed TxRef:

```typescript
type UniswapTxRef = Pick<TxRef, "blockNumber" | "blockHash" | "txHash">;
type NearTxRef = Omit<TxRef, "blockNumber" | "sender">;
```